reactor-core/src/main/java/reactor/core/publisher/DirectProcessor.java

```diff
@@ -166,13 +189,33 @@ public Emission emitError(Throwable t) {
 	}
 
 	@Override
-	public Emission emitNext(T t) {
+	public void emitNext(T value) {
+		Emission result = tryEmitNext(value);
+		if (result == Emission.FAIL_OVERFLOW) {
```

Comment:
nit: could use a `switch` for readability _and_ performance

Reply:
done